controllers/rabbitmqcluster_controller_test.go

```diff
@@ -918,23 +758,30 @@ var _ = Describe("RabbitmqclusterController", func() {
 				Expect(err).NotTo(HaveOccurred())
 				return service.Annotations
 			}, 3).Should(HaveKeyWithValue("anno-key", "anno-value"))
-			var sts *appsv1.StatefulSet
+
 			Eventually(func() map[string]string {
-				sts, _ = clientSet.AppsV1().StatefulSets(rabbitmqCluster.Namespace).Get(statefulSetName, metav1.GetOptions{})
+				sts, _ := clientSet.AppsV1().StatefulSets(rabbitmqCluster.Namespace).Get(statefulSetName, metav1.GetOptions{})
 				return sts.Annotations
 			}, 3).Should(HaveKeyWithValue("anno-key", "anno-value"))
 
-			// verify that SuccessfulUpdate events are recorded for all child resources
+			Eventually(func() map[string]string {
+				roleBinding, _ := clientSet.RbacV1().RoleBindings(rabbitmqCluster.Namespace).Get(rabbitmqCluster.ChildResourceName("server"), metav1.GetOptions{})
+				return roleBinding.Annotations
+			}, 3).Should(HaveKeyWithValue("anno-key", "anno-value"))
+
+			// verify that SuccessfulUpdate events are recorded for some child resources.K8s aggregates similar events if they are emitted in a very short period
+			// time.
 			allEventMsgs := aggregateEventMsgs(rabbitmqCluster, "SuccessfulUpdate")
 			Expect(allEventMsgs).To(ContainSubstring(fmt.Sprintf("updated resource %s of Type *v1.StatefulSet", rabbitmqCluster.ChildResourceName("server"))))
 			Expect(allEventMsgs).To(ContainSubstring(fmt.Sprintf("updated resource %s of Type *v1.Service", rabbitmqCluster.ChildResourceName("client"))))
 			Expect(allEventMsgs).To(ContainSubstring(fmt.Sprintf("updated resource %s of Type *v1.Service", rabbitmqCluster.ChildResourceName("headless"))))
 			Expect(allEventMsgs).To(ContainSubstring(fmt.Sprintf("updated resource %s of Type *v1.ConfigMap", rabbitmqCluster.ChildResourceName("server-conf"))))
+			Expect(allEventMsgs).To(ContainSubstring(fmt.Sprintf("updated resource %s of Type *v1.ConfigMap", rabbitmqCluster.ChildResourceName("plugins-conf"))))
 			Expect(allEventMsgs).To(ContainSubstring(fmt.Sprintf("updated resource %s of Type *v1.Secret", rabbitmqCluster.ChildResourceName("erlang-cookie"))))
 			Expect(allEventMsgs).To(ContainSubstring(fmt.Sprintf("updated resource %s of Type *v1.Secret", rabbitmqCluster.ChildResourceName("admin"))))
 			Expect(allEventMsgs).To(ContainSubstring(fmt.Sprintf("updated resource %s of Type *v1.ServiceAccount", rabbitmqCluster.ChildResourceName("server"))))
 			Expect(allEventMsgs).To(ContainSubstring(fmt.Sprintf("updated resource %s of Type *v1.Role", rabbitmqCluster.ChildResourceName("endpoint-discovery"))))
-			Expect(allEventMsgs).To(ContainSubstring(fmt.Sprintf("updated resource %s of Type *v1.RoleBinding", rabbitmqCluster.ChildResourceName("server"))))
+			Expect(allEventMsgs).To(ContainSubstring("combined from similar events"))
```

Comment:
Is this enough to make the test not flaky? I am not sure whether the problem will only happy for `v1.RoleBinding`. If this happens to `v1.Role` or any other types, we will still have flakes. And I would like the test to reliably pass in the future even if we create a new child resource.

The test here is to make sure that child resources annotations has been updated. I would suggest either to get these object to verify that updated annotations are there, or, to assert on the operator logs.